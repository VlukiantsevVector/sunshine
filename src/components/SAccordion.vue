<template>
  <div :class="classes()">
    <slot />
  </div>
</template>

<script>
import Vue from 'vue';
import ClassesMixin from './internal/ClassesMixin';

export default Vue.extend({
  name: 'SAccordion',

  inject: ['$validator'],

  mixins: [
    ClassesMixin,
  ],

  props: {
    linked: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    registerItem(item) {
      this.items.push(item);
    },

    notifyOpen(openedItem) {
      if (!this.linked) return;

      for (const item of this.items) {
        if (item !== openedItem) {
          item.close();
        }
      }
    },
  },

  created() {
    this.items = [];
  },
});
</script>
