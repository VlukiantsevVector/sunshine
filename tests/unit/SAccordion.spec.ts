import { shallowMount } from '@vue/test-utils';
import SAccordion from '@/components/SAccordion.vue';

describe('SAccordion', () => {
  it('renders item in slot', () => {
    const item = '<h1>Test</h1>';
    const wrapper = shallowMount(SAccordion, {
      slots: {
        default: item,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
