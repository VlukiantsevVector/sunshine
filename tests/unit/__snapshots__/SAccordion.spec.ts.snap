// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAccordion renders item in slot 1`] = `
<div class="s-accordion">
  <h1>Test</h1>
</div>
`;
